src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,297 @@
+import React, {useState, useRef} from "react";
+import Box from "../Box";
+import Typography from "../Typography";
+import Input from "../Input";
+import SvgIcon from "../SvgIcon";
+import Button from "../Button";
+import ClickAwayListener from "../ClickAwayListener";
+import {MenuList, MenuListHeading} from "../MenuList";
+import Popover from "../Popover";
+import {ListItem} from "../List";
+import Chip from "../Chip";
+import {faChevronDown, faSearch} from "@fortawesome/pro-regular-svg-icons";
+
+export interface DropdownProps {
+  label?: string;
+  placeholder?: string;
+  options?: any;
```

Comment:
Suggestion; can we define the interface for options?

Reply:
Example:

```
interface Option {
  id: string | number;
  text: string;
  selected?: boolean;
  category?: string;
  endAdornment?: React.ReactNode;
}
```

Reply:
correct me if i'm wrong. if we put an interface like so
`export interface MenuOption {
  startAdornment?: React.ReactNode;
  endAdornment?: React.ReactNode;
  onSelect?: (option?: MenuOption) => void;
  text?: string | number;
  children?: React.ReactNode;
  category?: string;
  style?: React.CSSProperties;
  "data-cy"?: string;
}`

would this restrict the option object? if we had to add something in the future, we would have to change the dropdown component?

Reply:
Typescript uses a structural type system, so additional attributes on the object won't cause an issue, as long as the mandatory attributes are provided — https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes.html#structural-type-system

Reply:
got it thanks!